zerver/forms.py

```diff
@@ -70,10 +71,13 @@ def check_subdomain_available(subdomain: str) -> None:
         raise ValidationError(error_strings['extremal dash'])
     if not re.match('^[a-z0-9-]*$', subdomain):
         raise ValidationError(error_strings['bad character'])
-    if len(subdomain) < 3:
+    if len(subdomain) == 0 and from_management_command:
+        raise ValidationError(error_strings['empty'])
+    if len(subdomain) < 3 and not from_management_command:
         raise ValidationError(error_strings['too short'])
     if is_reserved_subdomain(subdomain) or \
-       get_realm(subdomain) is not None:
+       get_realm(subdomain) is not None and \
+       not from_management_command:
```

Comment:
I think this logic with `and from_management_command` is hard to read.  You should just do an early return in the `from_management_command` case.